Sources/SwiftSDKGenerator/Queries/TarExtractQuery.swift

```diff
@@ -0,0 +1,46 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2022-202Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Helpers
+
+import struct SystemPackage.FilePath
+
+struct TarExtractQuery: CachingQuery {
+  let archivePath: FilePath
+  let destinationPath: FilePath
+  let outputBinarySubpath: [FilePath.Component]
+  let stripComponents: Int?
+
+  init(
+    file archivePath: FilePath,
+    into destinationPath: FilePath,
+    outputBinarySubpath: [FilePath.Component],
+    stripComponents: Int? = nil
+  ) {
+    self.archivePath = archivePath
+    self.destinationPath = destinationPath
+    self.outputBinarySubpath = outputBinarySubpath
+    self.stripComponents = stripComponents
+  }
```

Comment:
nit: IIUC the synthesized initializer will have a default argument for `stripComponents` if you mark it `var`

```suggestion
  var stripComponents: Int? = nil
```

Reply:
I added the explicit `init` to rename the parameters so they matched `untar(file: into: ...)` at the call site while still having meaningful names in the struct.  However the `run()` method isn't very complex so slightly less meaningful names are probably a fair trade off for removing the init.